app/uk/gov/hmrc/tai/model/domain/calculation/TotalTax.scala

```diff
@@ -58,6 +86,47 @@ case class IncomeCategory(
 
 object IncomeCategory {
   implicit val formats: OFormat[IncomeCategory] = Json.format[IncomeCategory]
+
+  val incomeCategorySeqReads = new Reads[Seq[IncomeCategory]] {
+    override def reads(json: JsValue): JsResult[Seq[IncomeCategory]] = {
+      val categoryNames =
+        Seq("nonSavings", "untaxedInterest", "bankInterest", "ukDividends", "foreignInterest", "foreignDividends")
+      val incomeCategoryList = incomeCategories(json, categoryNames)
+      JsSuccess(incomeCategoryList)
+    }
+  }
+
+  private def categoryTypeFactory(category: String) =
```

Comment:
Is this strictly speaking really a factory? It's more like a map really

Reply:
It was like this before. Again, I would not touch this yet.

Reply:
OK, we can leave for now